src/Raven.Server/Documents/Indexes/Persistence/Corax/AnonymousCoraxDocumentConverter.cs

```diff
@@ -45,21 +46,19 @@ public AnonymousCoraxDocumentConverter(Index index, bool storeValue = false)
         var scope = new SingleEntryWriterScope(_allocator);
         var storedValue = _storeValue ? new DynamicJsonValue() : null;
 
-
         foreach (var property in accessor.GetPropertiesInOrder(documentToProcess))
         {
             var value = property.Value;
 
             IndexField field;
-            if (knownFields.TryGetByFieldName(property.Key, out var binding))
+            try
             {
```

Comment:
Code inside the `try...catch` is not subject to any type of optimization. Tread with care there (there is also prolog and epilog), if you can avoid it the codegen of the whole method would benefit.

Reply:
Doen